grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java

```diff
@@ -84,7 +95,42 @@ public void handleResolvedAddressGroups(
 
     newLbAddressGroups = Collections.unmodifiableList(newLbAddressGroups);
     newBackendServers = Collections.unmodifiableList(newBackendServers);
-    grpclbState.handleAddresses(newLbAddressGroups, newBackendServers);
+    Map<String, Object> rawLbConfigValue = attributes.get(ATTR_LOAD_BALANCING_CONFIG);
+    Mode newMode = retrieveModeFromLbConfig(rawLbConfigValue, helper.getChannelLogger());
+    grpclbState.handleAddresses(newLbAddressGroups, newBackendServers, newMode);
+  }
+
+  @VisibleForTesting
+  static Mode retrieveModeFromLbConfig(
+      @Nullable Map<String, Object> rawLbConfigValue, ChannelLogger channelLogger) {
+    try {
+      if (rawLbConfigValue != null) {
```

Comment:
Inverting these conditions would make the code more clear avoiding lots of indentation:

```java
if (rawLbConfigValue == null) {
  return DEFAULT_MODE;
}
Object rawChildPolicies = rawLbConfigValue.get("childPolicy");
if (rawChildPolicies == null) {
  return DEFAULT_MODE;
}
...
```